internal/uvm/scsi.go

```diff
@@ -194,20 +232,55 @@ func (uvm *UtilityVM) AddSCSI(ctx context.Context, hostPath string, uvmPath stri
 // `uvmPath` is optional if a guest mount is not requested.
 //
 // `readOnly` set to `true` if the physical disk should be attached read only.
-func (uvm *UtilityVM) AddSCSIPhysicalDisk(ctx context.Context, hostPath, uvmPath string, readOnly bool) (*SCSIMount, error) {
-	return uvm.addSCSIActual(ctx, hostPath, uvmPath, "PassThru", false, readOnly)
+func (uvm *UtilityVM) AddSCSIPhysicalDisk(ctx context.Context, hostPath, uvmPath string, readOnly bool) (sm *SCSIMount, err error) {
+	sm, existed, err := uvm.allocateSCSIDevice(ctx, hostPath, uvmPath, VMAccessTypeIndividual)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		if err != nil {
+			uvm.deallocateSCSI(ctx, sm)
+		}
+	}()
+
+	if existed {
+		return sm, nil
+	}
+
+	err = uvm.addSCSIActual(ctx, sm, "PassThru", readOnly)
+	return sm, err
 }
 
 // AddSCSILayer adds a read-only layer disk to a utility VM at the next
 // available location and returns the path in the UVM where the layer was
 // mounted. This function is used by LCOW as an alternate to PMEM for large
 // layers.
-func (uvm *UtilityVM) AddSCSILayer(ctx context.Context, hostPath string) (*SCSIMount, error) {
+//
+// This is a read-only (LCOW) layer VHD. This function `must not` be used for Windows.
+func (uvm *UtilityVM) AddSCSILayer(ctx context.Context, hostPath string) (sm *SCSIMount, err error) {
 	if uvm.operatingSystem == "windows" {
 		return nil, ErrSCSILayerWCOWUnsupported
 	}
 
-	sm, err := uvm.addSCSIActual(ctx, hostPath, "", "VirtualDisk", true, true)
+	sm, existed, err := uvm.allocateSCSIDevice(ctx, hostPath, "", VMAccessTypeNoop)
```

Comment:
Let's add a comment here to the effect of "We include the controller and LUN in the UVM path for layer SCSI disks, but we don't know what they are until after calling `allocateSCSIDevice`. To work around this, we initially pass an empty string for the UVM path, then update it on the object below."